middleware/supress_notfound.go

```diff
@@ -16,11 +16,22 @@ func SupressNotFound(router *chi.Mux) func(next http.Handler) http.Handler {
 	return func(next http.Handler) http.Handler {
 		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 			rctx := chi.RouteContext(r.Context())
-			match := rctx.Routes.Match(rctx, r.Method, r.URL.Path)
+
+			// Match() updates rctx.
+			// Use a temporary route context to look ahead.
+			tctx := chi.NewRouteContext()
+
+			// rctx.Routes is the original *chi.Mux.
+			// This starts a new traversal from the top.
+
+			// Unfortunately, this makes valid paths with incorrect methods return 404 instead of 405.
+			// TODO: Identify invalid path vs valid path with invalid method.
```

Comment:
> FYI, this middleware returns 404 instead of 405 for requests with valid paths that have incorrect HTTP methods. Supporting 405 response would require a bigger change within the chi.Mux implementation.

@ezraisw Thanks for calling this out. Can we document this in the middleware's top-level comment and remove the TODO? I don't think it's worth the API change at this point.